doc/md/12-garbage-collection.md

```diff
@@ -0,0 +1,46 @@
+---
+sidebar_position: 12
+---
+
+# Garbage collection
+
+Garbage collection is an automatic process that manages memory usage by removing unreferenced or dead objects and freeing up allocated heap memory.
+
+Motoko’s default garbage collection method uses a copying approach that depends on the amount of heap memory currently in use. An alternative garbage collector uses a marking approach, which instead considers the amount of free heap memory. These garbage collectors are triggered when enough changes have been made to heap memory since the last collection cycle. Garbage collection can be forced to run after every message using the `--force-gc` flag in the project's `dfx.json` file:
+
+```json
+"defaults": {
+  "build": {
+    "packtool": "",
+    "args": "--force-gc"
+  }
+}
+```
+
+Both garbage collectors are limited by the ICP instruction limit per message, preventing them from collecting the entire heap memory pool in a single execution. As a result, [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters) cannot fully utilize the entire 4GiB memory pool, as some free space must be reserved for the garbage collector to operate.
```

Comment:
Both GCs? Which would be the other one? The incremental GC is not affected by this.
I think we should simply recommend using incremental GC if people have larger memory. Otherwise having this note looks quite bad for Motoko especially as we have a solution.